### PR TITLE
XML Parsing refactor:

### LINKED ISSUE #432: Refactor list objects API support
Need to improve XML parsing style (using namespace aware parsing functionality) to simplify code.

### PR DESCRIPTION
**Pull request by donatello**
Use namespace aware functions throughout and eliminate usage of large
repetitive constants.

Fixes #432.

### CODE REVIEW

File: minio/parsers.py [hunk 1]
```diff
@@ -141,6 +141,39 @@ def parse_list_buckets(data):
         raise InvalidXMLError('Invalid XML provided for "ListBucketsResult" '
                               'some fields are missing. Message: {0}'.format(error.message))
 
+def _parse_objects_from_xml_elts(bucket_name, contents, common_prefixes):
+    """Internal function that extracts objects and common prefixes from
+    list_objects responses.
+    """
+    objects = []
+    for content in contents:
+        key_elt = content.find('s3:Key', _S3_NS)
```
Comment (balamurugana) [inline minio/parsers.py]: Couple of thoughts.

1. cElementTree works with xpath ie `name = tree.find('volStatus/volumes/volume/volName').text`.

2. you could extend or have generic functions on top of  cElementTree classes to aware of namespace and returning urldecoded value by default.  You could avoid repeation of logic like `content.find("element", _S3_NS)`

File: minio/parsers.py [hunk 2]
```diff
@@ -141,6 +141,39 @@ def parse_list_buckets(data):
...
+        object_name = urldecode(key_elt.text) if key_elt != None else None
```
Comment (balamurugana) [inline minio/parsers.py]: You could tell me why we need to do take care of `None`? `urldecode(key_elt.text)` would throw `NoneType` exception which means the xml format is invalid.

Comment (donatello) [inline minio/parsers.py]: I did the change from the point-of-view of only refactoring without changing behaviour - the earlier version did the same.

Let me add a commit here that throws exceptions if certain fields are not present, after taking a look at the spec once again.

File: minio/parsers.py [hunk 3]
```diff
@@ -421,6 +380,8 @@ def _iso8601_to_localized_time(date_string):
     :param date_string: iso8601 formatted date string.
     :return: :class:`datetime.datetime`
     """
+    if date_string is None:
```
Comment (donatello) [inline minio/parsers.py]: Removed this check now.

File: minio/parsers.py (hunk 2 above)
Comment (donatello) [inline minio/parsers.py]: PR is updated.

File: minio/parsers.py (hunk 1 above)
Comment (donatello) [inline minio/parsers.py]: This is changed now, and addresses these points as well.

**Comment by donatello** [comment]
Closing in favour of #444

**Pull request closed without merging**